enhancements/installer/alibaba-cloud-ipi.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Alibaba Cloud IPI
+authors:
+  - "@lemondlut"
+reviewers:
+  - "@bradteng"
+  - "@rosawy"
+approvers:
+  - "@?"
+creation-date: 2021-03-25
+last-updated: 2021-03-25
+status: implementable
+
+---
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement covers adding support to install OpenShift clusters to Alibaba Cloud and details for the Installer to create infrastructure and run an OpenShift cluster on Alibaba Cloud.
+
+## Motivation
+
+Currently, openshift4.x does not support Alibaba Cloud via IPI installation. However, the feedback team got is that there is a lot of demand for this kind of installation.
+
+### Goals
+
+1. Installer creates infrastructure and installs OpenShift on Alibaba Cloud.
+
+### Non-Goals
+
+## Proposal
+
+### User Stories
+Enables `openshift-install` to create and destroy a cluster on Alibaba Cloud, which is managed and released and covered by CI tests.
+
+### Implementation Details/Notes/Constraints [optional]
+
+
+### Risks and Mitigations
+
+
+## Design Details
+
+### Infrastructure Global Configuration
+
+#### API
+Alicloud will need to be added [Alicloud Environments in the API](https://github.com/openshift/api/blob/master/config/v1/types_infrastructure.go#L339-L355):
+
+#### Cluster Config Operator
+Cluster config operator will [need to be added](https://github.com/openshift/cluster-config-operator/blob/master/pkg/operator/kube_cloud_config/) and the new API vendored in.
+
+#### Installer Manifests & Cloud Provider Config
+
+The [Alicloud cloud-provider-config](https://github.com/openshift/installer/blob/master/pkg/asset/manifests/alicloud) will need to be added.
+
+### Install Config
+New required fields will be added to `installconfig.platform.alicloud`.
```

Comment:
Worth mentioning a few of them if we already know which they are. E.g. I'd expect at least fields for `region`, something like `diskType` or `diskSizeGB` to customize disks, something like `vmType` or `vmSize` to customize the VM's to be used, something like `imageID` if users would like to manually specify the ID of the image to use, etc. 

It's also very common that users want to use some preexisting network resources instead of having the installer create one for them, for example, users might want to use an existing VNet. If that's a common case for your customers, we could add fields to point to these existing networking resources by their name, something like `vNet`, `controlPlaneSubnet` and `computeSubnet`, etc.